osgi.specs/docbook/160/service.jakartaws.xml

```diff
@@ -194,8 +216,23 @@ public class SOAPLoggingHandler implements SOAPHandler&lt;SOAPMessageContext&gt;
     
 }</programlisting>
 <para>
- Handlers will be added to the Endpoint in service ranking order, that means that Handlers with a higher service ranking will be called before those with a lower service ranking.
- If two handlers have the same ranking they are sorted according to their service id, having handlers registered later being called later in the handler chain. 
+ Handlers are assembled into a handler chain in service ranking order. This means that Handlers with a higher service ranking will be positioned before those with a lower service ranking in the chain.
```

Comment:
We need to discuss different types of handlers

```
When assembling a handler chain the implementation is required to group the handlers according to their types. Lower level, more protocol specific, handlers execute closer to the network layer and higher level handlers operate closer to the web service endpoint. For example with an incoming SOAP over HTTP request an `HttpHandler` would be called before a `SoapHandler`, and the `SoapHandler` would be called before a `LogicalHandler`.
Within a particular handler type the order of the handler chain is determined by service ranking order. This means that Handlers with a higher service ranking will be positioned earlier in the chain than those with a lower service ranking.
```

Reply:
I hope it is now clear enough?